### ISSUE TITLE
[BUG] CPU usage over 100% without any load introduced / images downloaded / containers running / VMs migrated

### ISSUE THREAD

**Issue by averyfreeman** [harvester/harvester#1700]
**Describe the bug**
Installed 2-node cluster about 30 minutes ago.  Management interface reports 102% CPU usage.  vSphere, 8 vCPUs (threads) allocated for each VM on E5-2650v3 and E5-2650v4 Supermicro whiteboxes with 16G ram each (128G ram total on each host).

**To Reproduce**
Install main node from ISO.
Install secondary node from ISO ("join cluster" option during startup)
Wait for secondary node to say "ready"
Go to management console Dashboard and see what we can do
Notice: Not much, cluster CPU already overloading
![chrome_quGRCzmxMD](https://user-images.githubusercontent.com/1308847/146546827-4a82f9a2-e9ba-44d0-90ff-54526fea2404.png)

Note: Logged out of management console and back in, now CPU reports 50% usage (still nothing running beyond self-loaded services):

![chrome_EUEgnkGLHE](https://user-images.githubusercontent.com/1308847/146546848-0986f054-e1a8-4c60-b2e8-0ec94feabc2f.png)

**Expected behavior**
Maybe be able to download an image or two from Docker hub (?)

**Support bundle**

Will add once I have URL for ticket

Update:  I started this process 5 minutes ago, looks like it just finished while I started writing this (when I came back after 5 min it still said 40% - must have jumped ahead pretty quickly, like 40 to 100 in 10% of the time).

From the look of Mazak-ui's post, there should have been a package for me to include. I came back after the support package bundler hung at 40% for a few minutes, hoping to get it to send to you.  It must have failed, but there was no visual feedback about its status.

Since it was the first time I had used the support bundler tool, I wasn't sure if it had behaved properly, such as something that sends automatically but has no "support sent" feedback, etc.  It looks like there was no "support failed" feedback, either.

Would be nice if there was a "Support Bundle Sent" modal that shows if bundle was successful, there was no visual feedback about whether or not it worked, I was just back at the menu for choosing either support bundle or kube config.

If there's a place in the filesystem this info is stored where I can grab it manually, please let me know.

I could get the kube config without issue, but doesn't look like there's much here:

```
apiVersion: v1
kind: Config
clusters:
- name: "local"
  cluster:
    server: "https://192.168.1.61/k8s/clusters/local"
    certificate-authority-data: "redacted"

users:
- name: "local"
  user:
    token: "kubeconfig-user-redacted"


contexts:
- name: "local"
  context:
    user: "local"
    cluster: "local"

current-context: "local"
```

**Environment:**
 - Harvester ISO version: 1.0-rc2 2021-12-17
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): vSphere on E5-2650v3 (20 thread) w/ 128GB DDR4 2133MHz (Primary), 3 other VMs running.  E5-2650v4 (24 thread) w/ 128GB DDR4 2133MHz (Secondary).

**Additional context**
High hopes for harvester, but if can't even demo in virtual environment, no chance of being able to migrate VM workload to it in the near future.

UPDATE - next day:

You can see here after a life of ~15 hours each, both hosts reporting this high level of CPU usage

![chrome_liT7RXz5wY](https://user-images.githubusercontent.com/1308847/146627087-c8debfd1-92d1-4bdb-a496-a760d1d75487.png)

I can't see the graph referenced by mazak-ui because I can't get that graph to load.  

Here's a snapshot of `/proc/stat` on the first harvester VM:

```
Every 1.0s: cat /proc/stat           harvester-chicken: Sat Dec 18 03:15:42 2021

cpu  2160402 9921 833180 41376937 71941 0 148328 0 0 0
cpu0 268679 1181 103718 5175246 8994 0 41280 0 0 0
cpu1 270478 1251 104180 5169483 8517 0 27006 0 0 0
cpu2 270702 1228 104170 5171583 9020 0 17990 0 0 0
cpu3 270032 1231 104039 5170660 9593 0 16608 0 0 0
cpu4 269612 1251 104598 5171733 8864 0 13881 0 0 0
cpu5 270753 1269 104187 5172130 9239 0 10161 0 0 0
cpu6 270126 1271 104224 5172314 8822 0 11252 0 0 0
cpu7 270006 1236 104054 5173509 8888 0 10148 0 0 0
intr 753661294 1 81 0 0 0 0 0 0 1 0 0 0 93 0 0 0 187 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 2500720 950678 1774333 704856 616278 1699016 1043477 3409609 1657438 0 82508 46 453 0 455 482 0 0 0 0 12232 0 0 0 0 0 0 0 0 0 0 0 0
. . . [ trunc: zeros ] . . .
```

Snapshot of `top` (notably reporting between 3% - 7% cpu usage - this was a high frame):

```
top - 03:17:44 up 15:40,  1 user,  load average: 0.75, 0.71, 0.86
Tasks: 514 total,   1 running, 513 sleeping,   0 stopped,   0 zombie
%Cpu(s):  6.6 us,  2.3 sy,  0.0 ni, 89.4 id,  1.1 wa,  0.0 hi,  0.6 si,  0.0 st
MiB Mem : 16002.30+total,  995.137 free, 7455.367 used, 7551.805 buff/cache
MiB Swap:    0.000 total,    0.000 free,    0.000 used. 8272.113 avail Mem

  PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
 2919 root      20   0 2836160 1.657g  60084 S 20.27 10.60 157:08.45 kube-apiserver
13102 root      20   0  821856 104232  33412 S 7.973 0.636  31:21.79 containerd
13145 root      20   0  900476 171756  68356 S 5.980 1.048  65:50.00 kubelet
27623 root      20   0  755636  52220  31448 S 4.983 0.319   0:00.15 kubectl
27617 root      20   0  755572  55368  30112 S 4.319 0.338   0:00.13 kubectl
 2783 root      20   0 11.218g 233124 109428 S 3.987 1.423  43:24.29 etcd
13899 root      20   0  736160  40728  20436 S 1.661 0.249   1:04.08 node-disk-manag
26210 root      20   0 2242680 141220  38228 S 1.661 0.862   0:59.11 virt-handler
26621 root      20   0 1367712 497772  79948 S 1.329 3.038  23:55.49 rancher
    1 root      20   0   93220  15248  10092 S 0.997 0.093   4:16.57 systemd
 3077 root      20   0  959028 162072  50856 S 0.997 0.989  12:29.08 kube-controller
 4685 root      20   0  771940  67676  42396 S 0.997 0.413   9:08.67 calico-node
 4042 root      20   0  716620  16700   8204 S 0.664 0.102   0:57.66 containerd-shim
16093 root      20   0 1043668 234064  43492 S 0.664 1.428   8:30.58 harvester
16850 10001     20   0  751024  58804  29684 S 0.664 0.359   4:18.23 adapter
 1323 root      20   0   12264   7420   1608 S 0.332 0.045   0:03.59 haveged
 6291 10001     20   0  754912  52000  27292 S 0.332 0.317   2:38.36 metrics-server
 9739 root      20   0  818144  85136  26548 S 0.332 0.520   0:47.69 webhook
10499 root      20   0 1332664 318840  44000 S 0.332 1.946  28:14.70 fleetagent
12927 root      20   0 1071916 215840  89692 S 0.332 1.317   5:31.80 rke2
14505 root      20   0  716736  11080   8040 S 0.332 0.068   0:09.16 containerd-shim
14877 root      20   0  747088  43928  28324 S 0.332 0.268   2:15.79 kube-vip
15396 root      20   0  819540  59416  29052 S 0.332 0.363   1:13.77 harvester-netwo
15399 root      20   0  751568  58060  29180 S 0.332 0.354   1:10.66 harvester-netwo
15984 root      20   0  814984 139840  30960 S 0.332 0.853   4:28.82 longhorn-manage
15985 1001      20   0 1648888 142760  36028 S 0.332 0.871   2:32.21 virt-operator
16743 nobody    20   0  737944  45128  24620 S 0.332 0.275   1:06.59 kube-state-metr
18862 root      20   0  742800  39216  26440 S 0.332 0.239   0:18.61 csi-attacher
19247 101       20   0  470976  66372   7064 S 0.332 0.405   0:07.93 nginx
23137 root      20   0 1904212  28548  13744 S 0.332 0.174   0:42.04 longhorn
23154 root      20   0 1535552  30680  14000 S 0.332 0.187   0:41.18 longhorn
26458 rancher   20   0   47376   4672   3668 R 0.332 0.029   0:00.09 top                                                                 2 root      20   0       0      0      0 S 0.000 0.000   0:00.04 kthreadd
```

`sysstat.service` is disabled, so no `mpstat`:

```
harvester-chicken:/var/log \# systemctl list-unit-files | grep stat
sysstat.service                         disabled        disabled
``` 

I enabled it, here's what `mpstat` and `sar` are reporting now:

```
# sar
Linux 5.3.18-59.37-default (harvester-chicken)  12/18/21        _x86_64_        (8 CPU)

03:35:49     LINUX RESTART      (8 CPU)
harvester-chicken:/var/log # mpstat
Linux 5.3.18-59.37-default (harvester-chicken)  12/18/21        _x86_64_        (8 CPU)

03:36:45     CPU    %usr   %nice    %sys %iowait    %irq   %soft  %steal  %guest  %gnice   %idle
03:36:45     all    4.84    0.02    1.87    0.16    0.00    0.33    0.00    0.00    0.00   92.77
harvester-chicken:/var/log # mpstat
Linux 5.3.18-59.37-default (harvester-chicken)  12/18/21        _x86_64_        (8 CPU)

03:36:50     CPU    %usr   %nice    %sys %iowait    %irq   %soft  %steal  %guest  %gnice   %idle
03:36:50     all    4.84    0.02    1.87    0.16    0.00    0.33    0.00    0.00    0.00   92.77
```

While exploring ways of monitoring resource usage on the host as a whole, the thought occurred to me:  Is there a chance the dashboard is reporting the CPU usage of a container, and not the host? 🤔

**Comment by gatesry** [comment]
**Describe the bug**
CPU reserved for 14/28 cores after 4 hours of idle. This is after a fresh bare-metal install on 3 Intel NUCs. Actual CPU usage is less then 10%.
![2021-12-17 12_37_19-Harvester and 2 more pages - Work - Microsoft​ Edge](https://user-images.githubusercontent.com/67442967/146592184-fe32d5e6-fd68-4c6b-b87b-5475f05ff740.png)
![2021-12-17 13_46_08-Harvester and 2 more pages - Work - Microsoft​ Edge](https://user-images.githubusercontent.com/67442967/146599591-1f0ae755-1969-4b9d-8c93-1deeadb6da2f.png)

**To Reproduce**
Steps to reproduce the behavior:
1. Grab latest RC from Harvester GH
2. Install on main node in UEFI, select static IP networking on 24 subnet, select static VIP mode, import SSH keys from GH.
3. Login and set password to dashboard.

**Expected behavior**
CPU reserved metrics would fall down to normal operating levels, as seen in previous releases.

**Support bundle**


[supportbundle_07b4ed9d-3d25-4c36-ba45-f7fe44781001_2021-12-17T19-51-05Z.zip](https://github.com/harvester/harvester/files/7737466/supportbundle_07b4ed9d-3d25-4c36-ba45-f7fe44781001_2021-12-17T19-51-05Z.zip)

**Environment:**
 - Harvester ISO version: v1.0.0-rc2
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):
 -  Intel NUC i7 10th gen w/ Samsung NVME 1 TB & 64GB RAM (first node)
 -  Intel NUC i7 10th gen w/ Samsung NVME 2 TB & 64GB RAM (second node)
 -  Intel NUC i5 5th gen w/ Samsung NVME 1 TB & 24GB RAM (third node)

**Additional context**

![2021-12-17 14_00_14-Harvester and 2 more pages - Work - Microsoft​ Edge](https://user-images.githubusercontent.com/67442967/146601122-accb77dc-9050-428b-9f6a-e780fef53239.png)


The problem persists in RC1. Have tried both UEFI and legacy GPT install methods. BIOS and NVME firmware is all up to date.

Thanks for any help troubleshooting.

**Comment by averyfreeman** [comment]
> **Describe the bug** CPU reserved for 14/28 cores after 4 hours of idle. This is after a fresh bare-metal install on 3 Intel NUCs. Actual CPU usage is less then 10%. ![2021-12-17 12_37_19-Harvester and 2 more pages - Work - Microsoft​ Edge](https://user-images.githubusercontent.com/67442967/146592184-fe32d5e6-fd68-4c6b-b87b-5475f05ff740.png) ![2021-12-17 13_46_08-Harvester and 2 more pages - Work - Microsoft​ Edge](https://user-images.githubusercontent.com/67442967/146599591-1f0ae755-1969-4b9d-8c93-1deeadb6da2f.png)

Oh interesting, so you measured actual CPU usage through `top`, `vmstat`, `watch -n 1 'cat /proc/stat'` or something, and it's just a UI issue? -- sorry as soon as I wrote that I noticed the graph in your 2nd picture.  Interesting, so they must use different methods of measurement.

It's nice to know the CPU utilization isn't so high.  Does the way it's being measured and misinterpreted on the first page of the dashboard create an inhibition of practical usage, or is it purely cosmetic?  

The high CPU usage reports scared me off before I investigated further, so I haven't even tried to do anything with the system yet - I was under the impression it was "accurate". It was also 5 AM in the morning, I had been up all night, and was finally was able to go to bed after filing the bug.

**Comment by gatesry** [comment]
@averyfreeman  - You got it, just reserved showing.

**Comment by averyfreeman** [comment]
> @averyfreeman - You got it, just reserved showing.

I updated OP quite a bit - I see it's your CPU usage graph, must be why I didn't notice / recognize it at first, mine won't load.   

That's very interesting that the VM's reserved CPU would be showing as consumed.  How did you know it correlates with reserved CPU, are you looking at numbers in vSphere?

Is there a way to run updates on this thing?  I notice there's `zypper` on the underlying system, but the filesystem is read-only.

**Comment by gatesry** [comment]
Try the newest 1.0.0 release and let me know if you still get the CPU issue. I'm reinstalling my cluster now and will report back.

Regarding the updates, I would advise not running manual updates on the Harvester OS. It's read-only I believe to prevent misconfigurations.

**Comment by TheEagleByte** [comment]
I am experiencing the same issue. There is nothing on these 3 servers. Fresh v1.0.0 install on all 3 servers:

![image](https://user-images.githubusercontent.com/28986398/160835981-b40abc9f-87c7-4a06-ae34-f525213ed9bd.png)

**Comment by ibrokethecloud** [comment]
@TheEagleByte the metric shows CPU reserved. Because harvester leverages k8s, we reserve some cpu/memory for the controlplane and core harvester infra components.

**Comment by ibrokethecloud** [comment]
@guangbochen we should improve the visualisation for the CPU / Memory resource usage.

One the hosts overview for example the CPU graph shows only Reserved details for the CPU/Memory.

![image](https://user-images.githubusercontent.com/19664014/171534577-e9db0e3c-b856-4088-ac53-27e70c5b6985.png)

While if we click through on the host we can see the more detailed breakdown of the Reserved vs Used utilization for the node in question. 
![image](https://user-images.githubusercontent.com/19664014/171534774-8dcca36b-94cf-44f8-9db3-85a64c07ab23.png)

We should overlay the two bar graphs on the overview screen, so say a different color can be used to indicate what amount of the Reserved CPU is actually Used.

This should make for a better UX.

**Comment by DacoDev** [comment]
﻿I recognize that the above issues seem to relate to the minimum reserved CPU that Harvester is looking for is roughly 8-9 cores, but I'm not sure if I should open a new bug, when I open the dashboard or the Host page the Reserved CPU amount rises unending, attached a screenshot where it was at 1354% or 54 cores, and I saw it was in the 5-digit % later on. No VMs or workloads of any sort, relatively fresh install of release 1.0.2.

![image](https://user-images.githubusercontent.com/20329637/171739030-453130fc-4427-485a-9374-a4dc674ea13e.png)

**Comment by ibrokethecloud** [comment]
Hi @DacoDev is there any chance if you could please click on the node name and provide a snapshot of the graphs when this happens?

**Comment by DacoDev** [comment]
Hello, on the Hosts tab, I see:
![image](https://user-images.githubusercontent.com/20329637/171763050-8862bc38-7bcd-4ee7-af92-1eeb4859c79f.png)

After clicking on the hostname I see:
![image](https://user-images.githubusercontent.com/20329637/171763032-66e2f037-963d-4c90-9019-76cafb242cfb.png)

I did try Chrome, Firefox, Brave, and Safari to be sure it wasn't an extension or runaway browser JS issue.

Here's the lines in the page source that are updating:
![image](https://user-images.githubusercontent.com/20329637/171763296-f31031b8-6f7c-44f0-b20c-f1685798d95e.png)

The gauge reset to about 4-5 of 4 C when I refresh the page and then starts counting up again, but it does not reset the count clicking between different pages without a refresh.

Let me know if I can provide anything else!

**Comment by ibrokethecloud** [comment]
Hi @DacoDev are you by any chance please able to generate a new support bundle?

**Comment by DacoDev** [comment]
> Hi @DacoDev are you by any chance please able to generate a new support bundle?

I have generated one, is there a better way to provide it than on the public issue? Not sure of any potentially sensitive logs or confs that could be included.

**Comment by ibrokethecloud** [comment]
i am around on rancher users slack

**Comment by ibrokethecloud** [comment]
Looks like the users node is 4cpu, and that is causing pod scheduling to fail.

That is likely to be the reason for the CPU spikes in this particular case.

**Comment by lanfon72** [comment]
The bug has been fixed on #2347.
